lhotse/cut/set.py

```diff
@@ -996,8 +996,7 @@ def subset(
                     f"CutSet has only {len(self)} items but last {last} required; not doing anything."
                 )
                 return self
-            cut_ids = list(self.ids)[-last:]
-            return CutSet.from_cuts(self[cid] for cid in cut_ids)
+            return CutSet.from_cuts(islice(self, len(self) - last, len(self)))
```

Comment:
Can you assign len(self) to a variable to avoid double iteration for lazy cut sets?

Reply:
Yeah, good point.

Reply:
Done.